lib/framedivider.rb

```diff
@@ -5,94 +5,216 @@
+  unless poly2 && (poly2.is_a?(OpenStudio::Point3dVector) || poly2.is_a?(Array))
+    msg = ""
+    msg << "'#{id2}': " unless id2.empty?
+    msg << "Invalid polygon (fits?, 2nd argument) - skipping"
+    TBD.log(TBD::DEBUG, msg)
+    return false
+  end
+  if poly1.empty?
+    msg = ""
+    msg << "'#{id1}': " unless id1.empty?
+    msg << "Empty polygon (fits?, 1st argument) - skipping"
+    TBD.log(TBD::DEBUG, msg)
+    return false
+  end
+  if poly2.empty?
+    msg = ""
+    msg << "'#{id2}': " unless id2.empty?
+    msg << "Empty polygon (fits?, 2nd argument) - skipping"
+    TBD.log(TBD::DEBUG, msg)
     return false
   end
 
-  pts.each do |pt|
+  poly1.each do |pt|
     unless pt.class == OpenStudio::Point3d
-      TBD.log(TBD::DEBUG,
-        "#{pt.class}? expected OSM 3D points to fit - skipping")
+      msg = "#{pt.class}? "
+      msg << "for '#{id1}': " unless id1.empty?
+      msg << "Expected OpenStudio 3D points (fits?, 1st argument - skipping"
+      TBD.log(TBD::DEBUG, msg)
+      return false
+    end
+  end
+  poly2.each do |pt|
+    unless pt.class == OpenStudio::Point3d
+      msg = "#{pt.class}? "
+      msg << "for '#{id2}': " unless id2.empty?
+      msg << "Expected OpenStudio 3D points (fits?, 2nd argument - skipping"
+      TBD.log(TBD::DEBUG, msg)
       return false
     end
   end
 
-  s = model.getSubSurfaceByName(id)
-  if s.empty?
-    TBD.log(TBD::DEBUG,
-      "Can't find '#{id}' OSM subsurface to fit - skipping")
+  ft = OpenStudio::Transformation::alignFace(poly1).inverse
+  ft_poly1 = flatZ( (ft * poly1).reverse )
+  ft_poly2 = flatZ( (ft * poly2).reverse )
+
+  ft_model = OpenStudio::Model::Model.new
+  surface1 = OpenStudio::Model::Surface.new(ft_poly1, ft_model)
+  surface2 = OpenStudio::Model::Surface.new(ft_poly2, ft_model)
+
+  area1 = surface1.grossArea
+  unless area1 > TOL
+    msg = ""
+    msg << "'#{id1}': " unless id1.empty?
+    msg << "Invalid polygon area (fits?, 1st argument) - skipping"
+    TBD.log(TBD::DEBUG, msg)
     return false
   end
-
-  s = s.get
-  if s.surface.empty?
-    TBD.log(TBD::DEBUG,
-      "Missing subsurface '#{id}' OSM parent surface entry - skipping")
+  area2 = surface2.grossArea
+  unless area2 > TOL
+    msg = ""
+    msg << "'#{id2}': " unless id2.empty?
+    msg << "Invalid polygon area (fits?, 2nd argument) - skipping"
+    TBD.log(TBD::DEBUG, msg)
     return false
   end
 
-  nom = s.surface.get.nameString
-  parent = model.getSurfaceByName(nom)
-  if parent.empty?
-    TBD.log(TBD::DEBUG,
-      "Can't find subsurface '#{id}' OSM parent surface '#{nom}' - skipping")
+  union = OpenStudio::join(ft_poly1, ft_poly2, TOL)
+  return false if union.empty?
+  union = union.get
+  surface = OpenStudio::Model::Surface.new(union, ft_model)
+  area = surface.grossArea
+  return false if area < TOL
+  if area1 > area2
+    return false if area - area1 > TOL
+  else
+    return false if area - area2 > TOL
+  end
```

Comment:
```suggestion
  if area1 > area2
    return false if Math.abs(area - area1) > TOL
  else
    return false if Math.abs(area - area2) > TOL
  end
```

Probably not needed?